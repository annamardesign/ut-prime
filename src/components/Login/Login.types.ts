import { WithStyles, withStyles, createStyles } from '@material-ui/core/styles';
import topLogo from './images/topLogo.png';
import bottomLogo from './images/bottomLogo.png';
import React from 'react';
import error from '../images/error.png';

export interface ILoginProps {
    authenticated: boolean;
    identityCheck: ({}) => Promise<{}>;
}

const styles = createStyles({
    loginContainer: {
        display: 'flex',
        position: 'absolute',
        top: '50%',
        flexDirection: 'column',
        alignItems: 'center',
        width: '100%',
        transform: 'translateY(-50%)'
    },
    loginLogo: {
        width: '480px',
        height: '80px'
    },
    loginPageHeader: {
        marginBottom: '55px',
        background: `url(${topLogo}) no-repeat center`
    },
    loginPageFooter: {
        marginTop: '55px',
        background: `url(${bottomLogo}) no-repeat center`
    },
    loginForm: {
        boxSizing: 'border-box',
        width: 480,
        padding: '30px 95px 55px',
        border: '1px solid #E9E9E9',
        backgroundColor: '#FFFFFF',
        boxShadow: '0 1px 5px 0 rgba(85,89,102, 0.25)'
    },
    formContainer: {
        width: '100%'
    },
    formBody: {
        height: '100%'
    },
    loginTitle: {
        color: '#626262',
        textAlign: 'center',
        marginBottom: 30,
        fontSize: 16
    },
    formError: {
        display: 'flex',
        width: '100%',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 30
    },
    errorMessage: {
        marginBottom: 5,
        lineHeight: '18px',
        textAlign: 'center',
        fontSize: 16,
        color: '#E84949'
    },
    errorIcon: {
        background: `url(${error}) no-repeat`,
        width: 40,
        height: 40
    }
});

export const Styled = withStyles(styles);
export type StyledType = React.FC<ILoginProps & WithStyles<typeof styles>>
