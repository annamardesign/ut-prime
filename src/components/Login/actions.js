import {LOGIN, BIO_LOGIN, COOKIE_CHECK, LOGOUT} from 'ut-front-react/containers/LoginForm/actionTypes';

const getTimezone = () => {
    const offset = (new Date()).getTimezoneOffset();
    const sign = offset > 0 ? '-' : '+';
    const hours = Math.floor(Math.abs(offset) / 60);
    const minutes = Math.abs(offset) - (hours * 60);
    return `${sign}${('00' + hours.toString()).substr(-2)}:${('00' + minutes.toString()).substr(-2)}`;
};

export const identityCheck = (params) => ({
    type: LOGIN,
    method: 'identity.check',
    suppressErrorWindow: true,
    params: Object.assign(params, {$http: {uri: '/login'}, timezone: getTimezone(), channel: 'web'})
});

export const bioScan = () => ({
    type: BIO_LOGIN,
    method: 'bio.scan',
    suppressErrorWindow: true
});

export const cookieCheck = (params) => ({
    type: COOKIE_CHECK,
    method: 'identity.check',
    suppressErrorWindow: true,
    params: Object.assign({channel: 'web'}, (params || {}))
});

export const logout = () => ({
    type: LOGOUT,
    method: 'identity.closeSession',
    suppressErrorWindow: true,
    params: {}
});
