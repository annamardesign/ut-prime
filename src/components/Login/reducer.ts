import {LOGIN, COOKIE_CHECK} from './actions';
import Immutable from 'immutable';

export default (state = null, action: { type: Symbol; methodRequestState: string; result: {language: {}}; }) => {
    if ([LOGIN, COOKIE_CHECK].includes(action.type) && action.methodRequestState === 'finished') {
        //! todo simplify
        return action.result ? Immutable.fromJS({
            result: action.result,
            language: action.result.language
        }) : false;
    }
    return state;
};
