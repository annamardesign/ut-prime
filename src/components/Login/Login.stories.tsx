import React from 'react';
import { withReadme } from 'storybook-readme';

// @ts-ignore: md file and not a module
import README from './README.md';
import Login from './index';

export default {
    title: 'Login',
    component: Login,
    decorators: [withReadme(README)],
    args: {
        state: {
            login: false
        }
    }
};

export const Basic: React.FC<{}> = () =>
    <Login />;
