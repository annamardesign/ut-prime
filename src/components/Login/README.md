# Login

Show a login form

## How to use

```jsx
import Login from 'ut-prime/core/Login';

<Login />
```

## Props

| propName  | propType | defaultValue | isRequired |
| --------- | -------- | ------------ | ---------- |
