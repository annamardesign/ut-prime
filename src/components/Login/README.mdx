# Login
import { Canvas, Story, ArgsTable } from "@storybook/addon-docs";

Show a login form

## How to use

```jsx
import Login from 'ut-prime/core/Login';

<Login />
```

<Canvas>
    <Story id="login--basic" />
</Canvas>

## Properties

<ArgsTable story="."/>
