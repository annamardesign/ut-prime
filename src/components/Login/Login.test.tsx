import React from 'react';

import { render } from '../test';
import Login from './index';

expect.addSnapshotSerializer({
    test: val => typeof val === 'string' && /^dx_dx-.*_username/.test(val),
    serialize: () => '"test-id"'
});

describe('<Login />', () => {
    it('render equals snapshot', () => {
        const {
            getByTestId
        } = render(<Login />, {
            login: null
        });
        expect(getByTestId('ut-front-test')).toMatchSnapshot();
    });
});
