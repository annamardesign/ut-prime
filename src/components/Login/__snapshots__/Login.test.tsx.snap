// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> render equals snapshot 1`] = `
<div
  data-testid="ut-front-test"
>
  <div
    class="Login-loginContainer-1"
  >
    <div
      class="Login-loginLogo-2 Login-loginPageHeader-3"
    />
    <div
      class="MuiBox-root MuiBox-root-12 Login-loginForm-5"
    >
      <div
        class="Login-loginTitle-8"
      >
        <span>
          System Login
        </span>
      </div>
      
      <form
        autocomplete="off"
        class="p-fluid Login-formContainer-6"
      >
        <div
          class="p-field p-float-label"
        >
          <input
            class="p-inputtext p-component"
            name="username"
          />
          <label>
            Username
          </label>
        </div>
        <button
          class="p-button p-component"
          type="submit"
        >
          <span
            class="p-button-label p-c"
          >
            Next
          </span>
        </button>
      </form>
    </div>
    <div
      class="Login-loginLogo-2 Login-loginPageFooter-4"
    />
  </div>
</div>
`;
